CONTRIBUTING.rst

```diff
@@ -1,35 +1,220 @@
+  - Start each sentence on a new line.
 
 - Other (file names, CLI, etc.): follow the style of the existing code base.
 
-.. _`Contributor License Agreement`: contributor_license.html
-.. _`cla-assistant`: https://github.com/cla-assistant/
-.. _`PEP 8`: https://www.python.org/dev/peps/pep-0008/
+
+Versions and releases
+---------------------
+
+- We use `semantic versioning <https://semver.org>`_.
+- We keep at least two active milestones on each of the ``message_ix`` and ``ixmp`` repositories:
+
+  - The next minor version. E.g. if the latest release was 3.5, the next minor release/milestone is 3.6.
+  - The next major version. E.g. 4.0.
+
+- The milestones are closed at the time a new version is released.
+  If a major release (e.g. 4.0) is made without the preceding minor release (e.g. 3.6), both are closed together.
+
+- Every issue and PR must be assigned to a milestone to record the decision/intent to release it at a certain time.
+
+- New releases are made by Energy Program staff using the `Release procedure <https://github.com/iiasa/message_ix/wiki/Release-procedure>`_, and appear on Github, PyPI, and conda-forge.
+
+
+Contributing tutorials
+----------------------
+
+Developers *and users* of the |MESSAGEix| framework are welcome to contribute **tutorials**, according to the following guidelines.
+Per the license and CLA, tutorials will become part of the ``message_ix`` test suite and will be publicly available.
+
+Developers **must** ensure new features (including ``message_ix.tools`` submodules) are fully documented.
+This can be done via the API documentation (this site) and, optionally, a tutorial.
+These have complementary purposes:
+
+- The API documentation, built using Sphinx and ReadTheDocs, must completely, but succintly, *describe the arguments and behaviour* of every class and method in the code.
+- Tutorials serve as *structured learning exercises* for the classroom or self-study.
+  The intended learning outcome for each tutorial is that students understand how the model framework API may be used for scientific research, and can begin to implement their own models or model changes.
+
+Coding & writing style
+~~~~~~~~~~~~~~~~~~~~~~
+
+- Tutorials are formatted as Jupyter notebooks in Python or R.
+- Commit 'bare' notebooks in git, i.e. without cell output.
+  Notebooks will be run and rendered when the documentation is generated.
+- Add a line to ``tests/test_tutorials.py``, so that the parametrized test function runs the tutorial (as noted at `#196 <https://github.com/iiasa/message_ix/pull/196>`_).
+- Optionally, use Jupyter notebook slide-show features so that the tutorial can be viewed as a presentation.
+- When relevant, provide links to publications or sources that provide greater detail for the methodology, data, or other packages used.
+- Providing the mathematical formulation in the tutorial itself is optional.
+- Framework specific variables and parameters or functions must be in italic.
+
```

Comment:
@khaeru, if needed we can mention:
"- Adding relevant figures and tables in the tutorial is recoemmnded if that helps the users to understand the concept more easily." And perhaps, you can add a note mentioning where these sumpplementary figures/tables should be loaded as raw files.